pkg/controller/useraccount/useraccount_controller_test.go

```diff
@@ -1,134 +1,163 @@
 package useraccount
 
 import (
+	"context"
+	"fmt"
 	"testing"
 
-	"github.com/stretchr/testify/require"
-
 	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
 	"github.com/codeready-toolchain/member-operator/pkg/apis"
+	"github.com/codeready-toolchain/member-operator/pkg/config"
+	userv1 "github.com/openshift/api/user/v1"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/kubernetes/scheme"
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 )
 
-func TestEnsureUser(t *testing.T) {
+func TestReconcileOK(t *testing.T) {
 	logf.SetLogger(logf.ZapLogger(true))
 	username := "johnsmith"
 	userID := "34113f6a-2e0d-11e9-be9a-525400fb443d"
 	s := scheme.Scheme
 	err := apis.AddToScheme(s)
 	require.NoError(t, err)
 
-	t.Run("create_user", func(t *testing.T) {
-		userAcc := newUserAccount(username, userID)
-		client := fake.NewFakeClient()
-		reconciler := &ReconcileUserAccount{
-			client: client,
-			scheme: s,
-		}
+	// given
+	userAcc := newUserAccount(username, userID)
 
-		newUser, created, err := reconciler.ensureUser(userAcc)
-		assert.NoError(t, err)
-		assert.True(t, created)
-		assert.NotNil(t, newUser)
-		assert.Equal(t, username, newUser.Name)
-	})
+	// Objects to track in the fake client.
+	objs := []runtime.Object{userAcc}
 
-	t.Run("get_user", func(t *testing.T) {
-		userAcc := newUserAccount(username, userID)
-		user := newUser(userAcc)
-		client := fake.NewFakeClient(user)
-		reconciler := &ReconcileUserAccount{
-			client: client,
-			scheme: s,
-		}
+	// Register operator types with the runtime scheme.
+	s.AddKnownTypes(toolchainv1alpha1.SchemeGroupVersion, userAcc)
 
-		newUser, created, err := reconciler.ensureUser(userAcc)
-		assert.NoError(t, err)
-		assert.False(t, created)
-		assert.NotNil(t, newUser)
-		assert.Equal(t, userAcc.Name, newUser.Name)
-	})
-}
+	client := fake.NewFakeClient(objs...)
```

Comment:
I think that you don't need the `objs` array - you can pass the object directly to the function:
```go
client := fake.NewFakeClient(userAcc)
```

Reply:
Addressed in https://github.com/codeready-toolchain/member-operator/pull/25/commits/81f9d57c0d73f11d0430e1ae57b28891038d1a96